programs/zstdcli.c

```diff
@@ -1306,10 +1302,12 @@ int main(int const argCount, const char* argv[])
             }
         }
 
-        if ((filenames->tableSize==1) && outFileName)
-          operationResult = FIO_compressFilename(prefs, outFileName, filenames->fileNames[0], dictFileName, cLevel, compressionParams);
-        else
-          operationResult = FIO_compressMultipleFilenames(prefs, filenames->fileNames, (unsigned)filenames->tableSize, outMirroredDirName, outDirName, outFileName, suffix, dictFileName, cLevel, compressionParams);
+        if ((filenames->tableSize==1) && outFileName) {
+            operationResult = FIO_compressFilename(prefs, outFileName, filenames->fileNames[0], dictFileName, cLevel, compressionParams);
+        } else {
+            FIO_setNbFiles(prefs, (int)filenames->tableSize);
```

Comment:
since `filenames->tableSize` is also provided as an argument of next function, `FIO_compressMultipleFilenames()`, would it be possible to skip the `FIO_setNbFiles()` stage altogether ?

Reply:
Yeah - however, in our discussion below it seems like we'll be moving any text output further down into the function:
```
FIO_compressFilename_srcFile(FIO_prefs_t* const prefs,
                             cRess_t ress,
                             const char* dstFileName,
                             const char* srcFileName,
                             int compressionLevel) 
```
which seems like if we want to pass a `nbFiles` and/or `currFileIdx` field, we'd have to modify either the function signature of `FIO_compressFilename_srcFile()` or add two fields to `FIO_prefs_t`. I think I prefer adding fields to `FIO_prefs_t` since I don't think `FIO_compressFilename_srcFile()` should need a `nbFiles` and/or `currFileIdx` argument, since it is an operation on a single file.

Reply:
As an alternative, and assuming you want to limit the amount of changes,
`FIO_setNbFiles()` could be invoked from _within_ `FIO_compressMultipleFilenames()`